resources/providers/awslib/ec2/volume.go

```diff
@@ -0,0 +1,28 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package ec2
+
+type EC2Volume struct {
+	VolumeId   string
+	InstanceId string
+	Region     string
+	awsAccount string
```

Comment:
Are those fields needed?
can't we just use the native AWS volume type instead of creating a new type? I understand we'll have fields we don't necessarily need.

Reply:
I also wondered about that.
The reason I added the struct is to avoid of using AWS SDK structs all over our source code. In my opinion, the AWS providers should encapsulate the SDK calls and return as well a struct defined by our team.
However, I am open to hear opinions and learn the best practices regarding that extra layer and if it brings more benefits than overhead.

Reply:
IMO, if customized fields are unnecessary and our focus is on refining the provided structure, I would stick with simplicity and use the AWS struct, though it may not be crucial.
Regarding the `Region` and `awsAccount` fields, are they being used?